.github/workflows/end-to-end-test.yml

```diff
@@ -47,7 +47,7 @@ jobs:
         if: github.event.action == 'labeled' && github.event.label.name == env.E2E_TEST_LABEL
         uses: actions/checkout@v3
         with:
-          ref: ${{ github.event.pull_request.head.ref }}
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
intended? i don't quite know what this is doing

Reply:
yes intended :) See the commit message for that one. 
In short rather than checking out the HEAD reference it's the exact head sha at the time of the labeled event. 
So for this PR the E2E build checked out 464dc05153d1a368ffc67e4de62414404f5a39e5

Reply:
thx